### DISCUSSION TITLE
Assessing Crowd Density Using Person Detection in Frigate

### DISCUSSION THREAD

**Discussion by dan246** [blakeblackshear/frigate#10153] [Ask A Question]
I am interested in evaluating crowd density as low, medium, or high based on person recognition, but my model does not include crowd density as a label. Can this be achieved in Frigate? I am looking for a way to indirectly measure crowd congestion by identifying individuals. Any suggestions or methodologies to implement this would be greatly appreciated

**Comment by NickM-27** [comment]
You can use the person counts that are sent out on mqtt / websocket to assess this.

**Comment by dan246** [comment]
> 您可以使用 mqtt / websocket 上發送的人數來評估這一點。

Is it possible for me to send an event after calculating the number of people in the camera without using these two options? Can labels be replaced, such as modifying objects in the model that do not need to be reported or detected to be low density, medium density, high density, and then reporting them?

**Comment by dan246** [comment]
> You can use the person counts that are sent out on mqtt / websocket to assess this.

Would it be simpler to use MQTT or WebSocket? I'm not very familiar with them. If possible, could you also explain the setup methods for each? Thank you very much.

**Comment by NickM-27** [comment]
> Can labels be replaced, such as modifying objects in the model that do not need to be reported or detected to be low density, medium density, high density, and then reporting them?

that's not how models work. They are only trained to detect specific things. You are talking about an image classification model not an object detection model. 

depends where you are trying to use this information. Going to move to a discussion

**Reply by dan246** [reply]
How should I use MQTT or WebSocket to implement this? Is modifying the config.yaml file enough, or are there additional steps required? I want to allow users to set the number of minutes to be judged through the API themselves. If this can be achieved, I would be very grateful.

**Comment by dan246** [comment]
Hello,
I'm currently working on a project using Frigate for person detection. My approach involves creating several labels (e.g., A, B, C) and assigning these labels to the detected persons. However, I feel there might be something off with this method, or I might be missing a crucial part of the process.

Here's what I've been doing:

Set up Frigate to detect persons.
Once a person is detected, assign them one of several predefined labels (A, B, C).
I'm not sure if this is the most effective way to label detected individuals or if there's a more optimized method to approach this. Has anyone else experienced similar issues or has suggestions on how to improve this process? I'm looking for advice on how to refine my approach to ensure accurate and efficient labeling.

Thank you for your time and assistance.
